src/dmd/parse.d

```diff
@@ -284,6 +284,8 @@ final class Parser(AST) : Lexer
         int inBrackets; // inside [] of array index or slice
         Loc lookingForElse; // location of lonely if looking for an else
     }
+    Lexer lexer;
+    alias lexer this;
```

Comment:
I'd rather not use the `alias this`, and simply prefix uses of the lexer with `lexer.`

Reply:
This is replacing the inheritance, the PR will be MUCH bigger without it.

Reply:
That’s basically what my PR did: https://github.com/dlang/dmd/pull/9475/files